### PR TITLE
2316 API V3 Geocode View

### PR DESCRIPTION
**Pull request by lainsworth8801**
#### What's this PR do?
Updated endpoint version -> v3;
monkey test new endpoints;
Modified seed tests accordingly.

#### How should this be manually tested?
1. Copy mapquest api key in Organization -> Settings -> Mapquest API Key;
2. In inventory list, select properties or taxlots desired for geocoding:
Screen Shot 2020-08-26 at 4 26 08 PM
3. Geocode selected properties:
Screen Shot 2020-08-26 at 4 26 19 PM
4. Geocoded results:
Screen Shot 2020-08-26 at 4 26 25 PM
5. Same for taxlots:
Screen Shot 2020-08-26 at 4 27 52 PM

#### What are the relevant tickets?
#2316

### CODE REVIEW

File: seed/tests/test_geocode_views.py [hunk 1]
```diff
@@ -45,13 +49,14 @@ def test_geocode_endpoint_base_with_prepopulated_lat_long_no_api_request(self):
...
         property.save()
 
-        url = reverse('api:v2:geocode-geocode-by-ids')
+        self.property_view_factory.get_property_view(state=property)
+
         post_params = {
-            'organization_id': self.org.pk,
-            'property_ids': [property.id],
-            'taxlot_ids': []
+            'property_view_ids': [property.id],
```
Comment (adrian-lara) [inline seed/tests/test_geocode_views.py:55]: I think we want the -View ID from the PropertyView created in L52.

File: seed/tests/test_geocode_views.py [hunk 2]
```diff
@@ -64,70 +69,90 @@ def test_geocode_confidence_summary_returns_summary_dictionary(self):
...
-            'property_ids': [
+            'property_view_ids': [
                 property_none.id,
                 property_high.id,
                 property_low.id,
                 property_manual.id,
                 property_missing.id
             ],
-            'tax_lot_ids': [
+            'taxlot_view_ids': [
```
Comment (adrian-lara) [inline seed/tests/test_geocode_views.py:155]: Similar to the other comment, I think we want the -View IDs for the -View objects created in the new lines above.

File: seed/views/v3/geocode.py [hunk 3]
```diff
@@ -123,7 +123,7 @@ def confidence_summary(self, request):
 
         if taxlot_view_ids:
             taxlot_views = TaxLotView.objects.filter(
-                id__in=taxlot_view_ids,
+                state_id__in=taxlot_view_ids,
```
Comment (adrian-lara) [inline seed/views/v3/geocode.py:126]: Let's undo these changes. We don't want to make changes to v3 to make the tests pass. Instead, let's make more changes to the tests to accommodate how the API endpoints were changed from v2 to v3.

File: seed/static/seed/js/services/geocode_service.js [hunk 4]
```diff
@@ -29,7 +29,7 @@ angular.module('BE.seed.service.geocode', [])
 
       geocode_factory.check_org_has_api_key = function (org_id) {
         var params = {organization_id: org_id};
-        return $http.get('/api/v2/geocode/api_key_exists', {
+        return $http.get('/api/v3/organizations/' + org_id + '/api_key_exists', {
```
Comment (macintoshpie) [inline seed/static/seed/js/services/geocode_service.js:32]: Getting a 404 with this service (tested loading the mapping page). I think we need to change this to `'/api/v3/organizations/' + org_id + '/geocode_api_key_exists/'`

File: seed/static/seed/js/services/geocode_service.js [hunk 5]
```diff
@@ -19,7 +19,7 @@ angular.module('BE.seed.service.geocode', [])
       };
 
       geocode_factory.confidence_summary = function (property_state_ids, taxlot_state_ids) {
-        return $http.post('/api/v2/geocode/confidence_summary/', {
+        return $http.post('/api/v3/geocode/confidence_summary/', {
```
Comment (macintoshpie) [inline seed/static/seed/js/services/geocode_service.js:22]: Looks like we're missing the org query param here, which is used for auth. We should be able to add just like the `geocode_by_ids` call above

File: seed/static/seed/js/services/geocode_service.js [hunk 6]
```diff
@@ -19,7 +19,7 @@ angular.module('BE.seed.service.geocode', [])
...
           property_ids: property_state_ids,
           tax_lot_ids: taxlot_state_ids
         }).then(function (response) {
```
Comment (macintoshpie) [inline seed/static/seed/js/services/geocode_service.js:25]: We changed the request payload to accept view ids rather than state ids to make the APIs more consistent. Happy to help with this if you have any questions as it requires a few changes on the FE

File: seed/static/seed/js/services/geocode_service.js [hunk 7]
```diff
@@ -6,7 +6,7 @@ angular.module('BE.seed.service.geocode', [])
       var geocode_factory = {};
 
       geocode_factory.geocode_by_ids = function (property_state_ids, taxlot_state_ids) {
-        return $http.post('/api/v2/geocode/geocode_by_ids/', {
+        return $http.post('/api/v3/geocode/geocode_by_ids/', {
           property_ids: property_state_ids,
           taxlot_ids: taxlot_state_ids
         }, {
```
Comment (macintoshpie) [inline seed/static/seed/js/services/geocode_service.js:12]: We changed the request payload to accept view ids rather than state ids to make the APIs more consistent. Happy to help with this if you have any questions as it requires a few changes on the FE

File: seed/static/seed/js/services/geocode_service.js (hunk 4 above)
Comment (lainsworth8801) [reply seed/static/seed/js/services/geocode_service.js:32]: So regarding this...I think I just changed all the "v2" in this .js file where in this case I wasn't working directly on check_org_has_api_key. Would you rather me change it back to v2 in case someone else is working the ticket related to this call?

File: seed/static/seed/js/services/geocode_service.js (hunk 5 above)
Comment (macintoshpie) [reply seed/static/seed/js/services/geocode_service.js:22]: Looks like we still need to add the organization_id query param in this request. I'm testing this by going to the inventory list page, selecting a property, then clicking "Geocode selected" and checking the result in the network tab of devtools.
